src/status_im/utils/contenthash.cljs

```diff
@@ -23,17 +23,34 @@
                             hash))]
       (str "0xe301" (hex/encode (b58/decode b58-hash))))))
 
-(defn decode [hex]
+(defn decode
+  "TODO properly decode the CID
+   extract the content-type using varint ns"
+  [hex]
   (when (and hex (not= hex "0x")
-             (string/starts-with? hex "0xe30101")
-             ;; TODO properly decode the CID
-             ;; extract the content-type using varint ns
              (= 78 (count hex)))
-    {:namespace :ipfs
-     :hash  (-> hex
-                (subs 10)
-                hex/decode
-                b58/encode)}))
+    (cond
+      (string/starts-with? hex "0xe40101")
+      ;; assuming the content type is 2 bytes
+      {:namespace :swarm
+       :hash (-> hex
+                 (subs 10)
```

Comment:
should be 8 ?

Reply:
there is additional `01` for content type

Reply:
yes 2 bytes for content type

Reply:
so it should be 12 `0xe40101fa01` ?

Reply:
no fa01 is just one kind of contenthash